LetsBTF.sol

```diff
@@ -0,0 +1,759 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.9;
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
+import "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";
+import "./royalties/ERC2981ContractWideRoyalties.sol";
+import "./EIP712Signature.sol";
+import "erc721a/contracts/extensions/ERC721AQueryable.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+error MintNotActive();
+error TransferIsLocked();
+error MaxSupplyReached();
+error MintFundsMismatch();
+error TooMuchEthSent();
+error AllCreditClaimed();
+error InsufficientAvailableCredit();
+error CreditClaimFailed();
+error NoAmountProvided();
+error NoCreditUsed();
+error NotApprovedOrOwner();
+error TokenDoesNotExist();
+error ContractsCantMint();
+error InsufficientContractBalance();
+error WithdrawFailed();
+error WithdrawPercentageIncorrect();
+error MintBatchTooHigh();
+error MustMintBatchMultiple();
+error CreditAlreadySet();
+error NothingToWithdraw();
+error InitialCreditNotSet();
+error NoTokenIdsGiven();
+
+/**
+ * @title The first NFT with "built-in" liquidity.
+ * @author letsbtf.com
+ *
+ *
+ * Upon minting, Chainlink VRF is used to generate a verifiably 
+ * random number of which the credit value is derived from. 
+ * Hundreds of simulations were run to determine the best model 
+ * for credit distribution, which is as follows 
+ * (note: these are weighted probabilities but because a random
+ * number is used there will be slight variances to the 
+ * actual percentage ratios once the total supply is minted):
+ *
+ * 19% chance of 0.5 eth credit
+ * 17.9% chance of 0.75 eth credit
+ * 50% chance of 1 eth credit
+ * 10% chance of 1.25 eth credit
+ * 3% chance of 1.50 eth credit
+ * .10% chance of 10 eth credit
+ *
+ *
+ * #LetsBTF
+ *
+ */
+contract LetsBTF is
+    ERC721AQueryable,
+    ReentrancyGuard,
+    VRFConsumerBaseV2,
+    ERC2981ContractWideRoyalties,
+    EIP712Signature,
+    AccessControlEnumerable
+{
+    /* =========== MAPPINGS & ARRAYS =========== */
+
+    mapping(uint256 => TokenContainer) public requestIdToTokenId;
+
+    // Don't use this mapping directly if you need to get 
+    // the totalCredit and/or claimedCredit
+    // for a token externally, use the helper functions 
+    // `getTotalCreditForToken`, `getClaimedCreditForToken`
+    // and `getAvailableCreditForToken`
+    mapping(uint256 => Credit) private tokenCredits;
+
+    /* =========== STRUCTS =========== */
+
+    struct Credit {
+        uint256 totalCredit;
+        int256 claimedCredit;
+        uint256 timestampLastPendingIncrease;
+        uint256 lastCreditClaimIndex;
```

Comment:
Change both `timestampLastPendingIncrease` and `lastCreditClaimIndex` to `uint128`.